Source/CBLParseDate.c

```diff
@@ -172,9 +172,19 @@ static int parseTimezone(const char *zDate, DateTime *p){
         return c!=0;
     }
     zDate++;
-    if( getDigits(zDate, 2, 0, 14, ':', &nHr, 2, 0, 59, 0, &nMn)!=2 ){
+    
+    if( getDigits(zDate, 2, 0, 14, 0, &nHr) != 1 ) {
         return 1;
     }
+    
+    if (*zDate == ':') {
+        zDate++;
+    }
+    
+    if( getDigits(zDate, 2, 0, 59, 0, &nMn) != 1 ) {
```

Comment:
The `2` parameter here tells `getDigits` to parse a 2-character number. But in the test case that fails, the numbers is `0000` which is 4 digits long.